server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -309,6 +327,9 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(LegacyESVersion.V_7_9_0)) {
             out.writeOptionalWriteable(indexingPressureStats);
         }
+        if (ShardIndexingPressureSettings.isShardIndexingPressureAttributeEnabled()) {
```

Comment:
Curios, should the version be checked here since this is a new property? `if (out.getVersion().onOrAfter(Version.V_1_1_0)) {`

Reply:
This is only checking the node level attribute and is version agnostic.  This is to identify if all nodes in the cluster understand the new attribute (feature), or it is a heterogeneous cluster where few nodes will not honour new fields over the wire (for serialization).

Reply:
This check works by iterating over the discovery nodes and returning ``true`` only if all the nodes in the cluster supports the attribute. [code-link](https://github.com/opensearch-project/OpenSearch/blob/feature/478_indexBackPressure/server/src/main/java/org/opensearch/index/ShardIndexingPressureSettings.java#L79-L87)

So in the example above with cluster having heterogeneous nodes, this check will return ``false`` and new fields will not be sent/received over the wire.

Reply:
:+1: thank you